src/utils/client.ts

```diff
@@ -530,20 +530,39 @@ export default class Client {
 
   // Logic custom APIs
   runCustomRequest(
-    method: string,
-    url: string,
-    userData: Record<string, any> | any[],
-    headers: Record<string, string>
+    payload:
+      | string
+      | {
+          method: string;
+          url: string;
+          data: Record<string, any> | any[];
+          headers: Record<string, string>;
+        },
+    fieldValues: { [key: string]: any } | null = null
   ) {
     const { userId } = initInfo();
-    const data = {
+    const data: any = {
       fuser_key: userId,
-      form_key: this.formKey,
-      method,
-      url,
-      user_data: userData,
-      headers
+      form_key: this.formKey
     };
+
+    if (typeof payload === 'string') {
+      Object.assign(data, {
+        name: payload
+      });
+    } else {
+      Object.assign(data, {
```

Comment:
why not just assign one-by-one?

data.method = payload.method;
data.url = payload.url;

etc.